src/store/polling.js

```diff
@@ -1,25 +1,50 @@
-import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
-import { POLLING_IMAGES } from './action-types';
+const POLLING_SUFFIX = 'POLLING';
 
-const initialState = {};
+const hasPollingData = (polling = {}) => {
+  return polling?.id && polling?.fetcher && polling?.condition;
+};
 
-const addImageToPoll = (state, { payload: { id } }) => ({
-  ...state,
-  [id]: 'BUILDING',
-});
+const ofPollingType = (type) => type.endsWith(`_${POLLING_SUFFIX}`);
 
-const removeImageToPoll = (state, { ids }) => {
-  const newState = state;
-  ids.forEach((id) => {
-    delete newState[id];
-  });
-  return newState;
-};
+export const polling = (pollHash = {}) => ({ dispatch }) => (next) => (
+  action
+) => {
+  const { type, payload, meta } = action;
+  if (ofPollingType(type) && hasPollingData(payload)) {
+    const { id, fetcher, condition, onEvent } = payload;
+    if (pollHash[id]) {
+      return;
+    }
+    pollHash[id] = true;
+    setTimeout(() => {
+      dispatch({
+        type: id,
+        payload: fetcher().then((resp) => {
+          pollHash[id] = false;
+          const [toContinue, stateName] = condition(resp);
+          const nextActions = onEvent?.[stateName] || [];
+          nextActions.forEach((action) => action(dispatch));
```

Comment:
@karelhala - please notice that I moved from `notifications` to `onEvent` that its an object where the key is the event name (`success`, `failure`, etc) and the value is an array of functions that take `dispatch` and use it to dispatch an action.

Is that OK?

Reply:
Yeah, this is much better!